synapse/storage/data_stores/main/events.py

```diff
@@ -1473,3 +1523,11 @@ def _update_backward_extremeties(self, txn, events):
                 if not ev.internal_metadata.is_outlier()
             ],
         )
+
+    def _handle_redacted_unread_event_txn(self, txn: Connection, event: EventBase):
+        self.db.simple_update_txn(
```

Comment:
there's no guarantee that you've received the redacted event before the redaction; does this behave correctly?

also: does it behave correctly if it turns out that the sender of the redaction didn't have permission to redact the event (noting that that cannot be enforced when the redaction is received because again: the redaction might be received over federation before the redacted event).

I am strongly minded to ignore redactions in the first pass. if it causes too many cases of "I got a ghost unread notification" then we can fix it later?

Reply:
Yeah fair enough.